src/PKSim.Core/Snapshots/Mappers/SimulationMapper.cs

```diff
@@ -228,17 +242,28 @@ public override async Task<ModelSimulation> MapToModel(SnapshotSimulation snapsh
       {
          _logger.AddDebug(PKSimConstants.Information.LoadingSimulation(snapshot.Name), project.Name);
 
+         //Local cache of ids' that will be used to retrieve original building block parameters as the project is only registered 
+         //in global context once the whole snapshot mapping process is completed
+
+         var withIdRepository = new WithIdRepository();
+         var registrationVisitor = new RegisterObjectVisitor(withIdRepository);
+
          var simulation = await createSimulationFrom(snapshot, project);
 
          simulation.Solver = await _solverSettingsMapper.MapToModel(snapshot.Solver);
          simulation.OutputSchema = await _outputSchemaMapper.MapToModel(snapshot.OutputSchema);
          simulation.OutputSelections = await _outputSelectionsMapper.MapToModel(snapshot.OutputSelections, simulation);
 
-         await updateParameters(simulation, snapshot.Parameters);
+         registrationVisitor.Register(simulation);
```

Comment:
I need to do this here so that I can fnd the corresponding buildign block parameter. (in the copy of the bb saved in the simulation)